src/model.cpp

```diff
@@ -248,13 +248,16 @@ Status Model::reloadVersions(std::shared_ptr<model_versions_t> versionsToReload,
     for (const auto version : *versionsToReload) {
         SPDLOG_INFO("Will reload model: {}; version: {} ...", getName(), version);
         config.setVersion(version);
+
+        auto modelVersion = getModelInstanceByVersion(version);
+        if (modelVersion != nullptr) {
+            config.setLocalPath(modelVersion->getModelConfig().getLocalPath());
+        }
         auto status = config.parseModelMapping();
         if ((!status.ok()) && (status != StatusCode::FILE_INVALID)) {
             SPDLOG_ERROR("Error while parsing model mapping for model {}", status.string());
         }
 
-        auto modelVersion = getModelInstanceByVersion(version);
-
         if (!modelVersion) {
```

Comment:
Why not to move this if statement to 253 line instead?

Reply:
If we move this if statement to 253 line we would not get status, because `config.parseModelMapping()` returns one.

Reply:
Then why not moving parseModelMapping before `auto modelVersion = getModelInstanceByVersion(version);` ?